engine/src/execution_graph/logic_controllers/LogicalFilter.cpp

```diff
@@ -1,111 +1,218 @@
+#include <cudf/table/table_view.hpp>
+#include <cudf/hashing.hpp>
+#include <cudf/join.hpp>
+#include <cudf/stream_compaction.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
 
 #include "LogicalFilter.h"
-#include "cudf/stream_compaction.hpp"
-#include "cudf/hashing.hpp"
-#include "cudf/join.hpp"
 #include "../../CalciteExpressionParsing.h"
 #include "../../JoinProcessor.h"
-namespace ral{
+#include "../../Interpreter/interpreter_cpp.h"
 
-namespace processor{
+namespace ral {
+namespace processor {
+namespace {
 
-  const std::string INNER_JOIN = "inner";
-  const std::string LEFT_JOIN = "left";
-  const std::string RIGHT_JOIN = "right";
-  const std::string OUTER_JOIN = "full";
+const std::string LOGICAL_FILTER = "LogicalFilter";
+
+const std::string INNER_JOIN = "inner";
+const std::string LEFT_JOIN = "left";
+const std::string RIGHT_JOIN = "right";
+const std::string OUTER_JOIN = "full";
+
+} // namespace
+
+bool is_logical_filter(const std::string & query_part) {
+  return query_part.find(LOGICAL_FILTER) != std::string::npos;
+}
 
 std::unique_ptr<ral::frame::BlazingTable> applyBooleanFilter(
   const ral::frame::BlazingTableView & table,
-  const CudfColumnView boolValues){
-    auto filteredTable = cudf::experimental::apply_boolean_mask(
-      table.view(),boolValues);
-    return std::make_unique<ral::frame::BlazingTable>(std::move(
-      filteredTable),table.names());
+  const CudfColumnView & boolValues){
+  auto filteredTable = cudf::experimental::apply_boolean_mask(
+    table.view(),boolValues);
+  return std::make_unique<ral::frame::BlazingTable>(std::move(
+    filteredTable),table.names());
+}
+
+std::unique_ptr<cudf::column> evaluate_expression(
```

Comment:
this function should not exist. it should be the same function that is used for project and retuns a unique_ptr<BlazingTable>

Reply:
@rommelDB is working on the logical project operator so maybe this function needs to be deleted in his PR and replaced for the one used in logical project